components/nimbus/src/lib.rs

```diff
@@ -207,17 +220,43 @@ impl NimbusClient {
 
     pub fn apply_pending_experiments(&self) -> Result<Vec<EnrollmentChangeEvent>> {
         log::info!("updating experiment list");
+        // If the application did not pass in an installation date,
```

Comment:
Added this logic here because it happens off the main thread... Not sure if it's the best place for it